src/hermit/mod.rs

```diff
@@ -28,6 +28,547 @@ pub type wctype_t = i64;
 pub type regoff_t = size_t;
 pub type off_t = c_long;
 
+pub type sa_family_t = u8;
+pub type socklen_t = u32;
+pub type in_addr_t = u32;
+pub type in_port_t = u16;
+pub type time_t = i64;
+pub type suseconds_t = i64;
+pub type nfds_t = usize;
+
+s! {
+/// `timespec` is used by `clock_gettime` to retrieve the
+/// current time
+pub struct timespec {
+    /// seconds
+    pub tv_sec: i64,
+    /// nanoseconds
+    pub tv_nsec: i64,
```

Comment:
We should change this to `i32`, though this is a breaking change.

```suggestion
    pub tv_nsec: i32,
```

All other platforms define this as `long`, which is at least 32 bits.